R/plot-cumulative-time-profile.R

```diff
@@ -99,15 +99,43 @@ plotCumulativeTimeProfile <- function(data = NULL,
 
   # And optional color palette otherwise use colors from theme
   if (!isEmpty(plotConfiguration$colorPalette)) {
-    try(suppressMessages(
-      plotObject <- plotObject +
-        ggplot2::scale_fill_brewer(
-          palette = plotConfiguration$colorPalette,
-          aesthetics = c("color", "fill")
-        )
-    ))
+    if (isIncluded(plotConfiguration$colorPalette, .ViridisPalettes)) {
+      suppressMessages(
+        plotObject <- plotObject +
+          ggplot2::scale_fill_viridis_d(
+            option = plotConfiguration$colorPalette,
+            aesthetics = c("color", "fill")
+          )
+      )
+    } else {
+      # For unknown color palettes, ggplot2 throw a warning and default to Greens color palette
+      # Use tryCatch to use a better default (Set1) and suggest a colorPalette from enum ColorPalette
+      tryCatch(
+        {
+          # Silence ggplot2 message from using 'scale_fill_brewer':
+          # 'Adding another scale for fill, which will replace the existing scale'
+          suppressMessages(
+            plotObject <- plotObject +
+              ggplot2::scale_fill_brewer(
+                palette = plotConfiguration$colorPalette,
+                aesthetics = c("color", "fill")
+              )
+          )
+        },
+        warning = function(w) {
+          warning(messages$unknownColorPalette(plotConfiguration$colorPalette), call. = FALSE)
+          plotConfiguration$colorPalette <- ColorPalettes$Set1
+          suppressMessages(
+            plotObject <- plotObject +
+              ggplot2::scale_fill_brewer(
+                palette = plotConfiguration$colorPalette,
+                aesthetics = c("color", "fill")
+              )
+          )
+        }
+      )
+    }
   }
```

Comment:
@pchelle 
it seems that the code here and in plot-tornado is the same.
Can you extract it into a function (e.g. `applyColorPaletteTo(plotObject, plotConfig) `)?